numpy/lib/function_base.py

```diff
@@ -175,6 +175,16 @@ def histogram(a, bins=10, range=None, normed=False, weights=None,
             raise AttributeError(
                 'max must be larger than min in range parameter.')
 
+    # Histogram is an integer or a float array depending on the weights.
+    if weights is None:
+        ntype = np.dtype(np.intp)
+    else:
+        ntype = weights.dtype
+
+    # We set a block size, as this allows us to iterate over chunks when
+    # computing histograms, to minimize memory usage.
+    BLOCK = 65536
```

Comment:
@juliantaylor - actually could you clarify what you mean by doing the keep computation inplace? I tried doing:

```python
keep = tmp_a >= mn
np.logical_and(keep,  tmp_a <= mx, out=keep)
```

but I don't see any improvements. Did you mean something else?

I also just checked with ``BLOCK=65536/4`` for different array sizes from 1 to 1e8 but couldn't see any for which this was faster than the current block size.